src/pudl/extract/excel.py

```diff
@@ -176,9 +176,19 @@ def __init__(self, ds):
 
     def process_raw(self, df, page, **partition):
         """Transforms raw dataframe and rename columns."""
-        self.cols_added = []
+        # df = self.add_data_maturity(df, page, **partition)
```

Comment:
I couldn't add this as a standard `process_raw` step due to a test, which seems to attempt to initialize an extractor but does not generate a df and thus fails with a `df.loc` (see error below).  This is technically fine because none of the child classes use this `process_raw` (they all have something bespoke about them)... but any ideas on how to fix this?

```
__________________ TestGenericExtractor.test_read_excel_calls __________________

mock_read_excel = <MagicMock name='read_excel' id='5847261648'>

    @staticmethod
    @patch("pudl.extract.excel.pd.read_excel")
    def test_read_excel_calls(mock_read_excel):
        """Verifies that read_excel method is called with expected arguments."""
        mock_read_excel.return_value = pd.DataFrame()

>       FakeExtractor("/blah").extract(year=[2010, 2011])

test/unit/extract/excel_test.py:84:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src/pudl/extract/excel.py:250: in extract
    newdata = self.process_raw(newdata, page, **partition)
src/pudl/extract/excel.py:179: in process_raw
    df = self.add_data_maturity(df, page, **partition)
src/pudl/extract/excel.py:188: in add_data_maturity
    df.loc[:, "data_maturity"] = maturity
../../mambaforge/envs/pudl-dev/lib/python3.10/site-packages/pandas/core/indexing.py:716: in __setitem__
    iloc._setitem_with_indexer(indexer, value, self.name)

...

E                               ValueError: cannot set a frame with no defined index and a scalar

../../mambaforge/envs/pudl-dev/lib/python3.10/site-packages/pandas/core/indexing.py:1615: ValueError